darwin: python: pandas 0.19.2 failing tests

## Issue description

`pandas` fails to install because tests fail. The failing test is the clipboard test that is meant to be disabled on darwin through [this section](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/pandas/default.nix#L66). This is not actually disabling the test but rather forcing it to fail, but this can be fixed with a small change.
```diff
@@ -77,7 +77,7 @@ in buildPythonPackage rec {
     # The flag `-w` provides the initial directory to search for tests.
     # The flag `-A 'not network'` will disable tests that use internet.
     nosetests -w $out/${python.sitePackages}/pandas --no-path-adjustment -A 'not slow and not network' --stop \
-      --verbosity=3
+      --verbosity=3 --exclude test_clipboard
      runHook postCheck
   '';
```

*Then* we reach a different failing test: `integer_arithmetic`
```
test_integer_arithmetic (pandas.tests.test_expressions.TestExpressions) ... FAIL

======================================================================
FAIL: test_integer_arithmetic (pandas.tests.test_expressions.TestExpressions)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/c9ixmgqvqwk3iw4sq241ivgidxywbhcg-python2.7-pandas-0.19.2/lib/python2.7/site-packages/pandas/tests/test_expressions.py", line 107, in test_integer_arithmetic
    assert_frame_equal)
  File "/nix/store/c9ixmgqvqwk3iw4sq241ivgidxywbhcg-python2.7-pandas-0.19.2/lib/python2.7/site-packages/pandas/tests/test_expressions.py", line 100, in run_arithmetic_test
    assert_func(expected, result)
  File "/nix/store/c9ixmgqvqwk3iw4sq241ivgidxywbhcg-python2.7-pandas-0.19.2/lib/python2.7/site-packages/pandas/util/testing.py", line 1313, in assert_frame_equal
    obj='DataFrame.iloc[:, {0}]'.format(i))
  File "/nix/store/c9ixmgqvqwk3iw4sq241ivgidxywbhcg-python2.7-pandas-0.19.2/lib/python2.7/site-packages/pandas/util/testing.py", line 1181, in assert_series_equal
    obj='{0}'.format(obj))
  File "pandas/src/testing.pyx", line 59, in pandas._testing.assert_almost_equal (pandas/src/testing.c:3783)
  File "pandas/src/testing.pyx", line 173, in pandas._testing.assert_almost_equal (pandas/src/testing.c:2987)
  File "/nix/store/c9ixmgqvqwk3iw4sq241ivgidxywbhcg-python2.7-pandas-0.19.2/lib/python2.7/site-packages/pandas/util/testing.py", line 1018, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: DataFrame.iloc[:, 0] are different

DataFrame.iloc[:, 0] values are different (85.65143 %)
[left]:  [-9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, 4, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, 437893890380859392, 27, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, 46656, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, 10000000000, -9223372036854775808, 302875106592253, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, 3125, -9223372036854775808, 285311670611, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, 16777216, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, 823543, -9223372036854775808, -9223372036854775808, -9223372036854775808, 387420489, -9223372036854775808, -9223372036854775808, -9223372036854775808, 387420489, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, 823543, -9223372036854775808, -9223372036854775808, 46656, 437893890380859392, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, -9223372036854775808, ...]
[right]: [9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 4, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 437893890380859392, 27, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 46656, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 10000000000, 9223372036854775807, 302875106592253, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 3125, 9223372036854775807, 285311670611, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 16777216, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 823543, 9223372036854775807, 9223372036854775807, 9223372036854775807, 387420489, 9223372036854775807, 9223372036854775807, 9223372036854775807, 387420489, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 823543, 9223372036854775807, 9223372036854775807, 46656, 437893890380859392, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, 9223372036854775807, ...]

----------------------------------------------------------------------
Ran 7286 tests in 339.589s

FAILED (SKIP=233, failures=1)
```

I'm not sure if this is darwin specific, nixpkgs specific, or a known bug in `pandas`.
 
### Steps to reproduce

```shell
$ nix-build "<nixpkgs>" -A pythonPackages.pandas
```

## Technical details

* System: OSX Sierra 10.12.4 (16E195)
* Nix version: 1.11.6
* Nixpkgs version: 17.03.1052.aa0383397a

--- a/pkgs/development/python-modules/pandas/default.nix
+++ b/pkgs/development/python-modules/pandas/default.nix
@@ -19,7 +19,6 @@
 , openpyxl
 , tables
 , xlwt
-, darwin ? {}
 , libcxx ? null
 }:
 
@@ -53,7 +52,7 @@ in buildPythonPackage rec {
     openpyxl
     tables
     xlwt
-  ] ++ optional isDarwin darwin.locale; # provides the locale command
+  ];
 
   # For OSX, we need to add a dependency on libcxx, which provides
   # `complex.h` and other libraries that pandas depends on to build.
@@ -63,16 +62,22 @@ in buildPythonPackage rec {
     substituteInPlace setup.py \
       --replace "['pandas/src/klib', 'pandas/src']" \
                 "['pandas/src/klib', 'pandas/src', '$cpp_sdk']"
-
-  # disable clipboard tests since pbcopy/pbpaste are not open source
-    substituteInPlace pandas/io/tests/test_clipboard.py \
-      --replace pandas.util.clipboard no_such_module \
-      --replace OSError ImportError
   '';
 
   checkPhase = ''
     runHook preCheck
-    py.test $out/${python.sitePackages}/pandas --skip-slow --skip-network
+  ''
+  # TODO: Get locale and clipboard support working on darwin.
+  #       Until then we disable the tests.
+  + optionalString isDarwin ''
+    # Fake the impure dependencies pbpaste and pbcopy
+    echo "#!/bin/sh" > pbcopy
+    echo "#!/bin/sh" > pbpaste
+    chmod a+x pbcopy pbpaste
+    export PATH=$(pwd):$PATH
+  '' + ''
+    py.test $out/${python.sitePackages}/pandas --skip-slow --skip-network \
+      ${if isDarwin then "-k 'not test_locale and not test_clipboard'" else ""}
     runHook postCheck
   '';
 
@@ -83,7 +88,7 @@ in buildPythonPackage rec {
     homepage = "http://pandas.pydata.org/";
     description = "Python Data Analysis Library";
     license = stdenv.lib.licenses.bsd3;
-    maintainers = with stdenv.lib.maintainers; [ raskin fridh ];
+    maintainers = with stdenv.lib.maintainers; [ raskin fridh knedlsepp ];
     platforms = stdenv.lib.platforms.unix;
   };
 }

